lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -100,10 +109,25 @@ public ParquetReader(
             AggregatedMemoryContext systemMemoryContext,
             ParquetReaderOptions options)
             throws IOException
+    {
+        this(fileCreatedBy, messageColumnIO, blocks, nCopies(blocks.size(), null), dataSource, timeZone, systemMemoryContext, options);
+    }
+
+    public ParquetReader(
+            Optional<String> fileCreatedBy,
+            MessageColumnIO messageColumnIO,
+            List<BlockMetaData> blocks,
+            List<Long> blockStarts,
```

Comment:
This could be

```
            List<@Nullable Long> blockStarts,
```

but I'd rather make this `Optional<List<Long>`. Then it would be natural to just change the constructor, not need for overload. The old version of the ctor is called in one place only.

Reply:
Also, blockStarts -> blocksFirstRowIndex

Reply:
I'll call it `Optional<List<Long>> firstRowsOfBlocks`.